system/jlib/jmetrics.hpp

```diff
@@ -114,26 +107,25 @@ class jlib_decl MetricBase : public IMetric
     virtual const std::string &queryName() const override { return name; }
     virtual const std::string &queryDescription() const override { return description; }
     virtual MetricType queryMetricType() const override { return metricType; }
-    void addMetaData(MetricMetaData _metaData) { metaData.emplace_back(_metaData); }
-    void setMetaData(const std::vector<MetricMetaData> &_metaData) { metaData = _metaData; }
-    const std::vector<MetricMetaData> &queryMetaData() const { return metaData; }
+    const MetricMetaData &queryMetaData() const { return metaData; }
     StatisticMeasure queryUnits() const override { return units; }
 
 
 protected:
     // No one should be able to create one of these
-    MetricBase(const char *_name, const char *_desc, MetricType _metricType, StatisticMeasure _units) :
+    MetricBase(const char *_name, const char *_desc, MetricType _metricType, StatisticMeasure _units, const MetricMetaData &_metaData) :
```

Comment:
I slightly wonder about the efficiency of this - e.g. whether there should be a version with a && parameter.  I think leave for the moment, but might be worth revisiting at some point.